extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -202,22 +266,198 @@ export class AmpStory360 extends AMP.BaseElement {
     container.appendChild(this.canvas_);
     this.applyFillContent(container, /* replacedContent */ true);
 
-    return Services.storyStoreServiceForOrNull(this.win).then(
-      (storeService) => {
-        storeService.subscribe(
-          StateProperty.PAGE_SIZE,
-          this.resizeRenderer_.bind(this),
-          false /* callToInitialize */
-        );
-      }
+    this.element.getAttribute('controls') === 'gyroscope' &&
+      this.checkGyroscopePermissions_();
+
+    this.initializeListeners_();
+  }
+
+  /** @private */
+  initializeListeners_() {
+    this.storeService_.subscribe(
+      StateProperty.GYROSCOPE_PERMISSION_STATE,
+      (permissionState) => this.onPermissionState_(permissionState)
+    );
+
+    this.storeService_.subscribe(StateProperty.PAGE_SIZE, () =>
+      this.resizeRenderer_()
     );
   }
 
+  /**
+   * @param {string} permissionState
+   * @private
+   */
+  onPermissionState_(permissionState) {
+    if (permissionState === 'granted') {
+      this.enableGyroscope_();
+    } else if (permissionState === 'denied') {
+      this.gyroscopeControls_ = false;
+      this.togglePermissionClass_(true);
+    }
+  }
+
+  /** @private */
+  checkGyroscopePermissions_() {
+    //  Browser doesn't support DeviceOrientationEvent API.
+    if (typeof DeviceOrientationEvent === 'undefined') {
+      return;
+    }
+
+    // If browser doesn't require permission, enable gyroscope.
+    if (typeof DeviceOrientationEvent.requestPermission === 'undefined') {
+      this.enableGyroscope_();
+    }
+
+    // If DeviceOrientationEvent and permissions, build permission button.
+    if (typeof DeviceOrientationEvent.requestPermission === 'function') {
+      this.buildPermissionButton_();
+    }
+  }
+
+  /**
+   * Creates a device orientation listener and sets gyroscopeControls_ state.
+   * If listener is not called in 1000ms, remove listener and resume animation.
+   * @private
+   */
+  enableGyroscope_() {
+    this.gyroscopeControls_ = true;
+    this.togglePermissionClass_(true);
+
+    this.win.addEventListener('deviceorientation', (e) => {
+      this.isReady_ && this.onDeviceOrientation_(e);
+      clearTimeout(checkNoMotion);
+    });
+
+    const checkNoMotion = setTimeout(() => {
+      this.gyroscopeControls_ = false;
+      this.animate_();
```

Comment:
Only if `this.isReady_` and `this.isPlaying_`

Can we try to think of a fallback and introduce it in a follow up PR? We'd have to think of what's the behavior we want and carefully look at the code.
If you're super confident that this fallback is great feel free to pushback!

Reply:
I'll updated the comment to clarify this.
This happens on desktop browsers that support `deviceorientation` but aren't moving.
Removing this will prevent desktop safari and chrome from animating.

Handheld devices will call the listener even with a slight amount of motionso it will not fallback if the device is handheld.

From a UX perspective I think this is a necessary solution because we want desktop users to experience the animation, otherwise it will be a still image. From an engineering perspective I think it's an okay fallback to build on.